controller.py

```diff
@@ -0,0 +1,58 @@
+from flask import Blueprint, request, jsonify
+from service import FacultyService
+
+faculty_service = FacultyService()
+faculty_blueprint = Blueprint('faculty', __name__)
+
+
+@faculty_blueprint.route('/faculty', methods=['GET'])
+def get_all_faculties():
+    faculties = faculty_service.get_all_faculties()
+    return jsonify(faculties), 200
+
+
+@faculty_blueprint.route('/faculty/<int:faculty_id>', methods=['GET'])
+def get_faculty(faculty_id):
+    faculty = faculty_service.get_faculty_by_id(faculty_id)
+    if faculty:
+        return jsonify(faculty), 200
+    return jsonify({"error": "Faculty not found"}), 404
+
+
+@faculty_blueprint.route('/faculty', methods=['POST'])
+def create_faculty():
+    data = request.get_json()
+    if not data or 'name' not in data or 'department' not in data or 'title' not in data:
```

Comment:
here you have to convert request to dto object using pydantic or marshmallow to automatically map incoming JSON payloads to Python